src/app/services/fetch-data.service.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable } from '@angular/core';
+import { Http, Headers, RequestOptions, Response } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+import { Parser } from '../shared/parser';
+
+@Injectable()
+export class FetchDataService {
+
+  constructor(private http: Http) { }
+
+  fetchData(URL: string, data: any): Observable<any> {
+    console.log('[http service] fetch');
+    let url = URL;
+    let classPath = Parser.parseClassPath(url);
+    let headers = new Headers({ 'Content-Type': 'text/plain' });
+    let options = new RequestOptions({ headers: headers });
+    console.log(url);
+
+    switch (classPath) {
+      case 'signature': {
+        let body = 'compound=' + data.compound;
+        return this.http.post(url, body, options)
+          .map(res => ({'data': res.json(), 'type': classPath}))
```

Comment:
i thought about that when implementing - what i think is that parsing data into variables/arrays inside the service will make it more complex and will eventually spoil its dedicated character, which is to bring data via http from the remote server. Right?
Also, if the parsing was included inside the service we would need many getter methods to be defined, in order to gain access to the actual values - right now the whole object being return from the service contains all the data we need.
That's why we introduced a dedicated Parser to do the parsing process, so every component can use it locally and parse whichever values it needs from the response object.
This is clearly a personal design choice, if you believe it's more clear to do all the data-stuff inside the service let me know so to get it updated the way you wish.